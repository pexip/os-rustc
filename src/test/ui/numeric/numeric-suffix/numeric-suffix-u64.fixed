// run-rustfix

fn foo<N>(_x: N) {}

fn main() {
    foo::<u64>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42_u64);
    foo::<u64>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<u64>(42u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
}
