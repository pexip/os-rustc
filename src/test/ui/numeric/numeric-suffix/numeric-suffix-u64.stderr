error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:6:16
   |
LL |     foo::<u64>(42_usize);
   |                ^^^^^^^^ expected `u64`, found `usize`
   |
help: change the type of the numeric literal from `usize` to `u64`
   |
LL |     foo::<u64>(42_u64);
   |                   ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:10:16
   |
LL |     foo::<u64>(42_u32);
   |                ^^^^^^ expected `u64`, found `u32`
   |
help: change the type of the numeric literal from `u32` to `u64`
   |
LL |     foo::<u64>(42_u64);
   |                   ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:13:16
   |
LL |     foo::<u64>(42_u16);
   |                ^^^^^^ expected `u64`, found `u16`
   |
help: change the type of the numeric literal from `u16` to `u64`
   |
LL |     foo::<u64>(42_u64);
   |                   ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:16:16
   |
LL |     foo::<u64>(42_u8);
   |                ^^^^^ expected `u64`, found `u8`
   |
help: change the type of the numeric literal from `u8` to `u64`
   |
LL |     foo::<u64>(42_u64);
   |                   ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:19:16
   |
LL |     foo::<u64>(42_isize);
   |                ^^^^^^^^ expected `u64`, found `isize`
   |
help: change the type of the numeric literal from `isize` to `u64`
   |
LL |     foo::<u64>(42_u64);
   |                   ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:22:16
   |
LL |     foo::<u64>(42_i64);
   |                ^^^^^^ expected `u64`, found `i64`
   |
help: change the type of the numeric literal from `i64` to `u64`
   |
LL |     foo::<u64>(42_u64);
   |                   ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:25:16
   |
LL |     foo::<u64>(42_i32);
   |                ^^^^^^ expected `u64`, found `i32`
   |
help: change the type of the numeric literal from `i32` to `u64`
   |
LL |     foo::<u64>(42_u64);
   |                   ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:28:16
   |
LL |     foo::<u64>(42_i16);
   |                ^^^^^^ expected `u64`, found `i16`
   |
help: change the type of the numeric literal from `i16` to `u64`
   |
LL |     foo::<u64>(42_u64);
   |                   ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:31:16
   |
LL |     foo::<u64>(42_i8);
   |                ^^^^^ expected `u64`, found `i8`
   |
help: change the type of the numeric literal from `i8` to `u64`
   |
LL |     foo::<u64>(42_u64);
   |                   ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:34:16
   |
LL |     foo::<u64>(42.0_f64);
   |                ^^^^^^^^ expected `u64`, found `f64`
   |
help: change the type of the numeric literal from `f64` to `u64`
   |
LL |     foo::<u64>(42u64);
   |                  ~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-u64.rs:37:16
   |
LL |     foo::<u64>(42.0_f32);
   |                ^^^^^^^^ expected `u64`, found `f32`
   |
help: change the type of the numeric literal from `f32` to `u64`
   |
LL |     foo::<u64>(42u64);
   |                  ~~~

error: aborting due to 11 previous errors

For more information about this error, try `rustc --explain E0308`.
